classes/Mail.php

```diff
@@ -877,7 +884,15 @@ public static function toPunycode($to)
             return $to;
         }
 
-        return $address[0] . '@' . idn_to_ascii($address[1], 0, INTL_IDNA_VARIANT_UTS46);
+        if (defined('INTL_IDNA_VARIANT_UTS46')) {
+            return $address[0] . '@' . idn_to_ascii($address[1], 0, INTL_IDNA_VARIANT_UTS46);
+        }
+
+        if (version_compare(phpversion(), '8.0.0', '<')) {
```

Comment:
If the constant will be removed in a future PHP version, wouldn't it make more sense to check if the constant is defined instead?

Reply:
We can just add a comment before saying "will be removed in PHP 8, see http://..."